docs/reference/endpoints/imap.md

```diff
@@ -36,6 +36,22 @@ tls cert.crt key.key {
 
 See [TLS configuration / Server](/reference/tls/#server-side) for details.
 
+**Syntax**: proxy_protocol _trusted ips..._ { ... } <br>
+**Default**: not enabled
+
+Enable use of HAProxy PROXY protocol. Supports both v1 and v2 protocols.
+If a list of trusted IP addresses or subnets is provided, only connections
+from those will be trusted.
+
+TLS for the channel between the proxies and maddy can be configured
```

Comment:
Is `tls` top-level directive unused then?

Reply:
Oh, good catch! Yes, that's correct, [`inheritGlobal` is `false`](https://github.com/drdaeman/maddy/blob/3910d91be11145312a418560edb72c710301f6e3/internal/endpoint/imap/imap.go#L87) for the `tls` directive in `ProxyProtocol`. In my own setup, I've used different set of certificates and haven't realized global TLS config is a thing. Unless you think it's fine the way it is, I will work on a change that would make global config respected (in a same manner `tls` directive works for IMAP).

Reply:
In setupListeners in both SMTP and IMAP, created listener is wrapped using tls.NewListener using endpoint-level TLS config before proxy_protocol.NewListener is called. I believe proxy_protocol tls directive can be simply removed.